clair/clair.go

```diff
@@ -58,8 +59,23 @@ func (imageAnalysis ImageAnalysis) ShortName(l v1.Layer) string {
 //Config configure Clair from configFile
 func Config() {
 	uri = fmtURI(viper.GetString("clair.uri"), viper.GetInt("clair.port")) + "/v1"
-	host = viper.GetString("clair.host")
 	healthURI = fmtURI(viper.GetString("clair.uri"), viper.GetInt("clair.healthPort")) + "/health"
 	Report.Path = viper.GetString("clair.report.path")
 	Report.Format = viper.GetString("clair.report.format")
+
+	headersArray := viper.GetStringSlice("clair.headers")
+	for _, header := range(headersArray){
+		splitHeader := strings.Split(header, "=")
+		headers[splitHeader[0]] = splitHeader[1]
+	}
 }
+
+func SetRequestHeaders(request *http.Request) {
+	for name, value := range headers {
+		if strings.EqualFold(name, "host") {
```

Comment:
Have you tried without setting `request.Host`, but passing everything to `request.Header`?

`Host`is an HTTP header so it could work. if not, we should use instead

```yaml
clair:
  request:
    host: MYHOST
    headers:
      myHeader: header
```